osf/models/notable_domain.py

```diff
@@ -40,6 +45,56 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
+    @staticmethod
+    def check_resource_for_domains(obj, confirm_spam=False, send_to_moderation=False):
```

Comment:
I *think* we might be putting a bit more logic than necessary here in this static method because it needs to have a reference to the `obj` on which we extract domains. This is kinda antithetical to the idea of a static method, which shouldn't rely on data of a specific instance, but only on the relevant data of a class.

How about we slim down this static method a bit so that it only takes `content` and `guid` as parameters and `get_or_create` instances for `NotableDomain` and `DomainReference`? If we go this route, then we can define another `extract_domains` method in the `SpamMixin` that invokes this static method. The benefit of this solution is that a class method is always self-aware, so that we don't have to pass an `obj` around. We may also override the `extract_domains` behavior for each subclass of `SpamMixin` to have more granular control over domain extraction behavior.